File: draw.cpp
#include<iostream>
#include<vector>
#include<algorithm>


using namespace std;

void zerofunc(int pref[][300],int n,int p)
{

	for(int i=0;i<n;i++)
	{
		for(int j=0;j<n;j++)
		{
			if(pref[i][j]==p)
				pref[i][j]=0;
		}
	}
}



int main()
{
	
	int pref[300][300];
	int random[300],room[300];
	int n,m,p,q,r;
	cout<<"enter the number of students"<<endl;
	cin>>n;
	
	for(int i=0;i<n;i++)
	{
		if((i+1)%10==1)
			cout<<endl<<"enter the preference of "<< i+1<<"st student"<<endl<<endl;
		else if((i+1)%10==2)
			cout<<endl<<"enter the preference of "<< i+1<<"nd student"<<endl<<endl;
		else if((i+1)%10==3)
			cout<<endl<<"enter the preference of "<< i+1<<"rd student"<<endl<<endl;
		else
			cout<<endl<<"enter the preference of "<< i+1<<"th student"<<endl<<endl;
		for(int j=0;j<n;j++)
		{
			cout<<"enter the preference number "<<j+1<<" -- ";
			cin>>pref[i][j];
		}
	}
	for(int i=0;i<n;i++)
		random[i]=i+1;
	random_shuffle(random, random+n);
	random_shuffle(random,random+rand()%n);

	cout<<endl;
	for(int i=0;i<n;i++)
		cout<<random[i]<<" ";
	cout<<endl;

	for(int k=1;k<=n;k++)
	{
		for(int i=0;i<n;i++)
		{
			if(k==random[i])
			{
				for(int j=0;j<n;j++)
				{
					if(pref[i][j]>0)
					{
						room[i]=pref[i][j];
						r=room[i];
						zerofunc(pref,n,r);
						break;
					}
				}
			}
		}
	}

for(int i=0;i<n;i++)
{
	cout<<"student "<<i+1<<" alloted room number :"<<room[i]<<endl;
}
return 0;
}

File: draw.py
from random import shuffle
while (1):

	rooms = list()

	def zerofunc(t,r):
		global rooms
		for i in range(t):
			for j in range(t):
				if (r==rooms[i][j]):
					rooms[i][j]=0

	def draw():
		global rooms
		rand=list()
		# pref1=[]
		t=int(input("enter the number of students "))
		pref1=[0 for _ in range(t)]
		for i in range (t):
			rooms.append([])
			print("enter the preference for student number",i+1," ")
			for j in range (t):
				pref=int(input(''))
				rooms[i].append(pref)
		for i in rooms:
			print (i)

		for i in range(t):
			rand.append(i+1)
		shuffle(rand)
		for i in rand:
			print(i)

		for i in range (1,t+1):
			for j in range(t):
				if (i == rand[j]):
					for k in range(t):
						if (rooms[j][k] > 0):
							pref1[j]=rooms[j][k]
							#print(pref1)
							r=pref1[j]
							zerofunc(t,r)
							# print(rooms)
							break
		for i in range(0,t):
			print ("preference of student",i+1,"is",pref1[i])
	draw()
	inp1=int(input("want to continue: 1 for yes,0 for no "))
	if (inp1==0):
		break	

	
